lua/bufferline/render.lua

```diff
@@ -231,10 +238,10 @@ local function render(update_names)
   end
 
   -- To prevent the expansion of the last click group
-  result = result .. '%0@BufferlineMainClickHandler@'
+  result = result .. '%0@BufferlineMainClickHandler@' .. hl('BufferTabpageFill')
 
-  if layout.actual_width + 1 <= layout.buffers_width and len(items) > 0 then
-    result = result .. hl('BufferTabpageFill') .. icons.bufferline_separator_inactive
+  if layout.actual_width + len(icons.separator_inactive) <= layout.buffers_width and len(items) > 0 then
```

Comment:
If we're going to replace the `1` here probably `strwidth` is the best function, I wouldn't be surprised that people use multi-bytes unicode characters (eg nerdfont chars). OTOH I don't want to support icons of more than 1 length, and `1` avoids function calls, so maybe adding a comment would also work.